### DISCUSSION TITLE
Session integration

### DISCUSSION THREAD

**Discussion by amiranagram** [clue/framework-x#93] [Q&A]
Hi @clue, I'd like to contribute a session middleware.

I'm not sure where to start though. Is it gonna be a part of the framework, or a separate package? If it's the part of the framework, what would be the ideal way to structure it, according to you? Folders, namespaces, etc.

I'm coming from https://framework-x.org/docs/integrations/sessions/

**Comment by clue** [comment]
@amiranagram Sounds awesome, would love to see a session middleware that integrates nicely with X!

I think as a starting point, it's probably best to start with a separate, standalone package and then look into integration at a future point in time. This way, we can work out the details that work best for your specific requirements before integrating something into the core, which would potentially make it more rigid to avoid any BC concerns.

I've also built an OAuth implementation for X and would love to see how these things fit together in the long run.

**Reply by amiranagram** [reply]
Great. I have some free time this weekend, so I will start going at it.
